src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (date !== undefined) {
+    const valuesOfDate = date.split(fromFormat[3]);
+
+    let DD = 0;
+    let MM = 0;
+    let year = 0;
+    let result = '';
+
+    for (let i = 0; i <= fromFormat.length - 2; i++) {
+      switch (fromFormat[i]) {
+        case 'YYYY':
+          year = valuesOfDate[i];
+          break;
+        case 'YY':
+          year = valuesOfDate[i];
+          break;
+        case 'MM':
+          MM = valuesOfDate[i];
+          break;
+        case 'DD':
+          DD = valuesOfDate[i];
+          break;
+      }
+    }
```

Comment:
The loop to assign values for `DD`, `MM`, and `year` is more complex than necessary. Since you know the indexes of the date parts from the `fromFormat`, you can directly assign the values without looping.